src/bitcoin/electrum/mod.rs

```diff
@@ -204,24 +204,22 @@ impl Electrum {
         let reorg_common_ancestor = if !is_rescanning {
             let changeset = self.bdk_wallet.apply_connected_chain_update(chain_update);
             let mut changes_iter = changeset.into_iter();
-            loop {
-                match changes_iter.next() {
-                    Some((height, Some(_))) => {
-                        // `BlockHash` being `Some(_)` means a checkpoint at this height was added to the chain.
-                        // Since we iterate in ascending height order, we'll see the lowest block height first.
-                        // If the lowest height it adds is higher than our height before syncing, we're good.
-                        // Else if it's adding a block at height before syncing or lower, it's a reorg.
-                        break if height > local_chain_tip.height() {
-                            None
-                        } else {
-                            log::info!("Block chain reorganization detected.");
-                            // We can assume height is positive as genesis block will not have changed.
-                            Some(self.bdk_wallet.find_block_before_height(height))
-                        };
-                    }
-                    Some((_, None)) => continue,
-                    None => break None,
+            if let Some((height, _)) = changes_iter.next() {
```

Comment:
Makes sense. Anything inserted or removed at or before our current height should make us reorg, no need for this loop.